src/Serializable.ts

```diff
@@ -82,14 +99,47 @@ export class Serializable extends SerializableBase implements ISerializable {
         return that.postDeserialize(deserialized)
     }
 
+    /**
+     * `preDeserialize` can be overwritten to manipulate the value before the deserialization from string.
+     * This allows to add logic to the deserialization without having to override the deserialize method.
+     *
+     * This function will run before the {@link preFrom} method.
+     *
+     * @param value the object that will be manipulated before the actual parsing.
+     * @returns the manipulated object
+     */
     protected static preDeserialize(value: any): any {
         return value
     }
 
+    /**
+     * `postDeserialize` can be overwritten to manipulate the value after the deserialization from string.
+     * This allows to add logic to the deserialization without having to override the deserialize method.
+     *
+     * This function will run after the {@link postFrom} method.
+     *
+     * @param value the object that will be manipulated after the actual parsing.
+     * @returns the manipulated object
+     */
     protected static postDeserialize<T extends Serializable>(value: T): T {
         return value
     }
 
+    /**
+     * The main entrypoint of the parsing. This method be overwritten but the this context also has to be in the method signature.
+     * The recommended method for changing the parsing logic is to overwrite the `{@link preFrom}` and `{@link postFrom}` methods.
+     *
+     * This is an example on how to overwrite it:
+     * ```ts
+     * public static fromAny<T extends ClassThatExtendsSerializable>(this: Constructor<T>, value: IClassThatExtendsSerializable): T {
+     *   return (this as any).fromT(value)
+     * }
+     * ```
+     *
+     * @param this tells typescript that the context of this method is the current class
```

Comment:
@ param this tells typescript that the context of this method is the current class instance, which may be a subclass of Serializable